### PR TITLE
Hw pr6

### PR DESCRIPTION
**Pull request by kaptetos**

### CODE REVIEW

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+  - changeSet:
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml:2]: what's that?

File: src/main/java/com/example/springbookstore/config/SwaggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                .apis(RequestHandlerSelectors.basePackage("com.example.springbookstore.controller"))
+                .paths(PathSelectors.any())
+                .build();
+    }
+
+    @Bean
+    public UiConfiguration uiConfiguration() {
+        return UiConfigurationBuilder.builder().build();
+    }
+}
```
Comment (okuzan) [inline src/main/java/com/example/springbookstore/config/SwaggerConfig.java]:
```suggestion
}

```

File: src/main/java/com/example/springbookstore/config/SwaggerConfig.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger.web.UiConfiguration;
+import springfox.documentation.swagger.web.UiConfigurationBuilder;
+
+@Configuration
+public class SwaggerConfig {
+
```
Comment (okuzan) [inline src/main/java/com/example/springbookstore/config/SwaggerConfig.java]:
```suggestion
```

File: src/main/java/com/example/springbookstore/mapper/BookMapper.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.example.springbookstore.dto.BookDto;
+import com.example.springbookstore.dto.CreateBookRequestDto;
+import com.example.springbookstore.entity.Book;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Mappings;
+
+@Mapper(componentModel = "spring")
+public interface BookMapper {
+
```
Comment (okuzan) [inline src/main/java/com/example/springbookstore/mapper/BookMapper.java]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]
image
after your PR is approved, merge the code, delete the branch, and work only on one PR at a time, PRs in homeworks should contain only relevant changes, otherwise changes accumulate and it makes it hard to review

File: src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (kaptetos) [reply src/main/resources/db/changelog/db.changelog-master.yaml:2]: in previous tasks it was necessary to add Liquibase

File: src/main/java/com/example/springbookstore/entity/Book.java [hunk 5]
```diff
@@ -13,15 +14,77 @@ public class Book {
...
-
     private String coverImage;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/entity/Book.java]: It is better to use the Lombok's annotations `@Setter` and `@Getter`

File: src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (Ivan95kos) [reply src/main/resources/db/changelog/db.changelog-master.yaml:2]: You should add the tables for the entity, `books`, and other,
but not test_table

File: src/main/java/com/example/springbookstore/service/BookServiceImpl.java [hunk 6]
```diff
@@ -1,27 +1,77 @@
...
+    }
+
+    @Override
+    public void deleteBook(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book", id));
+
+        book.setDeleted(true);
+
+        bookRepository.save(book);
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/service/BookServiceImpl.java]: Implement a soft delete concept for the Book entity
and use the method `deleteById(id)`

File: src/main/java/com/example/springbookstore/entity/Book.java [hunk 7]
```diff
@@ -13,15 +14,77 @@ public class Book {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
-    private String title;
+    @Column(name = "deleted")
+    private boolean deleted = false;
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/entity/Book.java]:
```suggestion
    @Column(nullable = false)
    private boolean isDeleted = false;
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
left a few comments

File: src/main/java/com/example/springbookstore/service/BookServiceImpl.java [hunk 8]
```diff
@@ -1,27 +1,75 @@
...
+        Book updatedBook = bookRepository.save(book);
+
+        return bookMapper.entityToDto(updatedBook);
+    }
+
+    @Override
+    public void deleteBook(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book", id));
+
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/service/BookServiceImpl.java:65]: useless action
```suggestion
```

File: src/main/java/com/example/springbookstore/entity/Book.java [hunk 9]
```diff
@@ -1,27 +1,30 @@
...
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import lombok.Getter;
+import lombok.Setter;
 import java.math.BigDecimal;
 
 @Entity
+@Getter
+@Setter
 public class Book {
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/entity/Book.java:15]:
```suggestion
@SQLDelete(sql = "UPDATE books SET deleted = true WHERE id=?")
@Where(clause = "is_deleted=false")
@Table(name = "books")
public class Book {
```

File: src/main/java/com/example/springbookstore/repository/BookRepository.java [hunk 10]
```diff
@@ -1,7 +1,11 @@
 package com.example.springbookstore.repository;
 
 import com.example.springbookstore.entity.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+    Page<Book>findByTitleContainingIgnoreCase(String title, Pageable pageable);
+
```
Comment (Ivan95kos) [inline src/main/java/com/example/springbookstore/repository/BookRepository.java:10]: line
```suggestion
```

**Review by Ivan95kos — APPROVED** [review]
Great job, with a few minor comments

**Pull request merged**